src/iop/lowpass.c

```diff
@@ -519,7 +562,7 @@ void commit_params(struct dt_iop_module_t *self, dt_iop_params_t *p1, dt_dev_pix
   if(fabs(d->contrast) <= 1.0f)
   {
     // linear curve for contrast up to +/- 1
-    for(int k = 0; k < 0x10000; k++) d->ctable[k] = d->contrast * (100.0f * k / 0x10000 - 50.0f) + 50.0f;
+    for(int k = 0; k < 0x10000; k++) d->ctable[k] = d->contrast * (100.0f * k / 0x10000 - d->cfulcrum) + 50.0f;
```

Comment:
This is ok, but below...

Reply:
It should be ```d->ctable[k] = d->contrast * (100.0f * k / 0x10000 - d->cfulcrum) + d->cfulcrum;```

Reply:
Right I missed that!

Reply:
but then your function output is not properly bounded in [0; 1] : https://www.wolframalpha.com/input/?i=1.5+*+(x+-+0.25)+%2B+0.50,+x+%3D+%5B0,+1%5D

Reply:
> Makes the correction proportional to fulcrum, which is not what we look for.

it does not, the fulcrum is an offset. It's proportionnal to the contrast.

Reply:
...
https://www.wolframalpha.com/input/?i=1.+*+(x+-+0.5)+%2B+0.50,+x+%3D+%5B0,+1%5D

Reply:
imagine fulcrum = 0.25, the current function as you wrote it is not only adding contrast but also luminance, because it is shifted linearly by 0.25.